cal/cal/static/js/app.js

```diff
@@ -115,55 +115,54 @@ analyticsApp.component('tagList', {
   }
 });
 
-function TagsDetailCtrl($scope, $http) {
-  var tagUrl = '/v1/tags/' + this.tagId + '/events';
-  var eventweek = '/v1/tags/' + this.tagId + '/eventWeek';
-  var query_timezone = moment.tz.guess();
-  $scope.tagDetails = [];
-  $scope.tagEvents = [];
-  $scope.tagHours = this.tagHours;
-
-  $http({method: 'GET', url: tagUrl + '.json' }).
-  success(function successCallback(data) {
-    for (var i = 0; i < data.results.length; i++) {
-      var event = data.results[i];
-      $scope.tagEvents.push({
-        start: (new Date(event.start)).toString(),
-        name: event.name,
-      });
-    }
-  });
-  $http({
-    method: 'GET',
-    url: eventweek + '.json',
-    params: {
-      timezone: query_timezone,
-    }
-  }).
-  success(function successCallback(data) {
-    $scope.tagHours = $scope.tagHours / data.length;
+function TimelineBaseCtrl($scope, $http) {
```

Comment:
Okay, took me a while to figure out what's going on. So it looks like you're trying to find a way to reuse code between both Tags and ColorCategories by using `TimelineBaseCtrl` to set a function on the scope of both the TagDetail and CategoryDetail that both of these detail controllers can then uses to populate a scope variable that will be picked up by nvd3 to use in the graph.

Imagine if someone comes along later and sees this line `TimelineBaseCtrl.call(this, $scope, $http);`, which now does a bunch of stuff. They're not sure exactly why they need it, but somewhere deep inside a template file in another part of the codebase there was a reference to `ctrlDetails` which breaks if the TimelineBaseCtrl was not called.

In Angular, whenever you want to reuse functionality for your controllers, you're supposed to put that logic into a service. Try this architecture:

- Create a new `QueryService` that will represent things for any analytical component of our app that implements a query of events. i think this is a good name because both `Tag` and `ColorCategory` represent a way that you can query events on your calendars, and then use those queried queried events to run a number of different analyses like line graphs, cumulative stats, weekly stats, whatever. You may also have noticed that I've built Tags and ColorCategories to be extensible - it is very natural for us to implement arbitrary unions and intersections for these. For example, if I wanted to query for the times that I'm studying cs170 with Cindy or James, I can do `ColorCategory('schoolwork').intersection(Tag('James').union(Tag('Cindy')))`. I don't think stat is a good name because it is to general.
- Let this `QueryService` implement a function named something like `populateLineGraph`, which takes the same parameters you had but now instead of *mutating* a scope object that was passed in, return the data that you are currently assigning to `ctrlDetails`. Note: There are obviously many other cases where we could be using a service to simplify shared functionality between tags and categories but let's just only have your one function for now.
- In the TagDetails and CategoryDetails controllers, inject your new QueryService and use your new function to get the equivalent of the `ctrlDetails` data. Now you can pass this data into the `showGraph` function you had. If you need to refer to the data in the template, you can save it to a variable on the TagDetails or CategoryDetails controllers and reference it with `$ctrl.myDataVariable`. And all this will allow you to avoid scope soup.

For now I'll refrain from reviewing the rest as a lot is likely to change and I think it would be good to start anew.